### PR TITLE
Replaced atoi

### PR DESCRIPTION
**Pull request by akallabeth**
* Replaced ```atoi``` with ```strtol```, ```strtol```, ```_strtoui64``` and ```_strtoi64```
* Added range checks wherever possible
* Added conversion error checks wherever possible

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/common/assistance.c [hunk 1]
```diff
@@ -974,26 +964,26 @@ int freerdp_assistance_parse_file_buffer(rdpAssistanceFile* file, const char* bu
...
-
 		r = (char*) malloc(length + 1);
 
 		if (!r)
 			return -1;
 
 		CopyMemory(r, p, length);
 		r[length] = '\0';
+		errno = 0;
+		{
```
Comment (hardening) [inline libfreerdp/common/assistance.c]: Why are you setting errno here ?

File: libfreerdp/crypto/tls.c [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 #include <assert.h>
 #include <string.h>
+#include <errno.h>
```
Comment (hardening) [inline libfreerdp/crypto/tls.c]: not needed ?

File: libfreerdp/common/assistance.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/common/assistance.c]: to ensure no other function has set ERRNO (tried to do that consistently before calling  ```strto*```)

File: libfreerdp/crypto/tls.c (hunk 2 above)
Comment (akallabeth) [reply libfreerdp/crypto/tls.c]: correct, looks like a leftover

File: libfreerdp/common/assistance.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/common/assistance.c]: hmm, comment gone? Ah well, I tried to set ```errno = 0``` before ```strto*``` calls wherever possible

**Pull request merged**